lib/macho/macho_file.rb

```diff
@@ -519,87 +549,30 @@ def set_sizeofcmds(size)
       @raw_data[20..23] = size_raw
     end
 
-    # Updates the `name` field in a DylibCommand.
-    # @param dylib_cmd [MachO::DylibCommand] the dylib command
-    # @param old_name [String] the old dylib name
-    # @param new_name [String] the new dylib name
-    # @return [void]
-    # @private
-    def set_name_in_dylib(dylib_cmd, old_name, new_name)
-      set_lc_str_in_cmd(dylib_cmd, dylib_cmd.name, old_name, new_name)
-    end
-
-    # Updates the `path` field in an RpathCommand.
-    # @param rpath_cmd [MachO::RpathCommand] the rpath command
-    # @param old_path [String] the old runtime name
-    # @param new_path [String] the new runtime name
-    # @return [void]
-    # @private
-    def set_path_in_rpath(rpath_cmd, old_path, new_path)
-      set_lc_str_in_cmd(rpath_cmd, rpath_cmd.path, old_path, new_path)
-    end
-
-    # Updates a generic LCStr field in any LoadCommand.
-    # @param cmd [MachO::LoadCommand] the load command
-    # @param lc_str [MachO::LoadCommand::LCStr] the load command string
-    # @param old_str [String] the old string
-    # @param new_str [String] the new string
-    # @raise [MachO::HeaderPadError] if the new name exceeds the header pad buffer
+    # Inserts the given load command at the given offset.
+    # @param offset [Fixnum] the offset to insert at
+    # @param lc [MachO::LoadCommand] the load command to insert
+    # @param options [Hash]
+    # @option options [Boolean] :repopulate (true) whether or not to repopulate
+    #  the instance fields
     # @private
-    # @todo This needs to be replaced.
-    # @see https://github.com/Homebrew/ruby-macho/pull/35
-    def set_lc_str_in_cmd(cmd, lc_str, old_str, new_str)
-      new_sizeofcmds = header.sizeofcmds
-      old_str = old_str.dup
-      new_str = new_str.dup
-
-      old_prepad = cmd.class.bytesize + old_str.bytesize + 1
-      new_prepad = cmd.class.bytesize + new_str.bytesize + 1
-
-      # calculate the original and new padded sizes of the strings
-      old_padded_size = Utils.round(old_prepad, alignment)
-      new_padded_size = Utils.round(new_prepad, alignment)
-
-      # calculate the number of pad bytes used in the old and new strings
-      old_pad = old_padded_size - (old_prepad)
-      new_pad = new_padded_size - (new_prepad)
-
-      # pad the old and new strings with null bytes to meet command bounds
-      old_str << "\x00" * old_pad
-      new_str << "\x00" * new_pad
-
-      # calculate the new sizeofcmds in MH
-      new_sizeofcmds += new_padded_size - cmd.cmdsize
+    def insert_command(offset, lc, options = {})
+      context = LoadCommand::SerializationContext.context_for(self)
+      cmd_raw = lc.serialize(context)
+      new_sizeofcmds = sizeofcmds + cmd_raw.bytesize
 
-      if new_sizeofcmds + header.class.bytesize > low_fileoff
+      if header.class.bytesize + new_sizeofcmds > low_fileoff
```

Comment:
Actually, users should never hit this method directly anyways (since it's marked private both in the object and API), but doing sanity checks on `offset` is still a great idea.